src/pudl/workspace/resource_cache.py

```diff
@@ -3,43 +3,16 @@
 from abc import ABC, abstractmethod
 from pathlib import Path
 from typing import Any, NamedTuple
-from urllib.parse import urlparse
 
 import boto3
-import botocore
 import google.auth
-from botocore.client import Config
-from botocore.exceptions import ClientError
-from google.api_core.exceptions import BadRequest
-from google.api_core.retry import Retry
-from google.cloud import storage
-from google.cloud.storage.blob import Blob
-from google.cloud.storage.retry import _should_retry
-from google.resumable_media.common import DataCorruption
+from upath import UPath
 
 import pudl.logging_helpers
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
-def extend_gcp_retry_predicate(predicate, *exception_types):
```

Comment:
It would be nice if we could retain this more robust retry logic, but I didn't see an obvious way to force it into the `UPathCache` and we aren't depending on GCS by default any more, so I dropped it.

I retained the `AbstractCache` class since it defines the interface that's used by the `LayeredCache` as well as the `UPathCache` class.